CRM/Event/BAO/Query.php

```diff
@@ -362,7 +362,11 @@ public static function whereClauseSingle(&$values, &$query) {
         return;
 
       case 'participant_status_id':
-        if ($value && is_array($value) && strpos($op, 'IN') === FALSE) {
+        if ($value && !is_array($value) && strpos($value, ',')) {
```

Comment:
I think we can just call the generic    

     $query->_where[$grouping][] = CRM_Contact_BAO_Query::buildClause($field", $op, $value, $dataType);

& also the getQillForField for field here - for that latter one we need to do a little more work since it's on a class without a common parent. My vote is that we resolve the issue with CRM_Contact_BAO_Query not being able to share with all the others with a trait

Reply:
@eileenmcnaughton i'm not sure because it seems to cascade down into the general handling that happens in L409

Reply:
@seamuslee001 OK - I've pulled out & fixed here to be generic

https://github.com/civicrm/civicrm-core/pull/15803

I think the comma separated thing should be sorted earlier though so I left worrying about that out of scope until I've had a chance to check the case one